src/utils/helpers.js

```diff
@@ -73,3 +73,42 @@ export const flattenArray = arr =>
     (Array.isArray(item)
       ? [...acc, ...flattenArray(item)]
       : [...acc, item]), []).filter(item => !!item);
+
+// eslint-disable-next-line
+const sizeOfObject = (object, size) => {
+  if (object === null) {
+    return 0;
+  }
+
+  return Object.keys(object).reduce((bytes, key) => {
+    bytes += size(key);
+    try {
+      bytes += size(object[key]);
+    } catch (ex) {
+      if (ex instanceof RangeError) {
+        // circular reference detected, final result might be incorrect
+        // but we don't need to throw error
+        bytes = 0;
+      }
+    }
+
+    return bytes;
+  }, 0);
+};
+
+export const sizeOf = (object) => {
+  const sizes = {
+    string: 2,
+    boolean: 4,
+    number: 8,
+  };
+
+  if (typeof object === 'string') return object.length * sizes.string;
+  if (typeof object === 'boolean') return sizes.boolean;
+  if (typeof object === 'number') return sizes.number;
+  if (Array.isArray(object)) {
+    return object.map(sizeOf).reduce((acc, curr) => acc + curr, 0);
+  }
+  if (typeof object === 'object') return sizeOfObject(object, sizeOf);
+  return 0;
+};
```

Comment:
- 💅 AFAIK the data object of transaction type 0 is always a string. So I don't understand why you implemented all this extra logic for other data types that just ads dead code to our code base 🤔

Reply:
I just wasn't sure if it's gonna remain string. else we would call it "message" instead of data.

Reply:
AFAIK, there is no plan to change it in Lisk Core. I think it's called `data` just because that's how it was requested from the community in https://github.com/LiskHQ/lisk-sdk/issues/26 and nobody gave any thought to naming. 

I still don't see a good reason to keep all this extra logic in the repository, because one day it might be useful.

Reply:
agreed
I'll remove the other parts